rule/receiver-naming.go

```diff
@@ -3,24 +3,69 @@ package rule
 import (
 	"fmt"
 	"go/ast"
+	"strconv"
+	"strings"
+	"sync"
 
 	"github.com/mgechev/revive/internal/typeparams"
 	"github.com/mgechev/revive/lint"
 )
 
 // ReceiverNamingRule lints given else constructs.
-type ReceiverNamingRule struct{}
+type ReceiverNamingRule struct {
+	receiverNameMaxLength int
+	sync.Mutex
+}
+
+const defaultReceiverNameMaxLength = -1 // thus will not check
+
+func (r *ReceiverNamingRule) configure(arguments lint.Arguments) {
+	r.Lock()
+	defer r.Unlock()
+	if r.receiverNameMaxLength == 0 {
+		if len(arguments) < 1 {
+			r.receiverNameMaxLength = defaultReceiverNameMaxLength
+			return
+		}
+		arg := arguments[0]
+		argStr, ok := arg.(string)
+		if !ok {
+			panic(fmt.Sprintf("Invalid argument for %s rule. Expecting an string, got %T", r.Name(), arg))
+		}
+
+		parts := strings.Split(argStr, "=")
+		if len(parts) != 2 {
+			panic(fmt.Sprintf("Invalid argument for %s rule. Expecting an string of the form 'key=value', got %s", r.Name(), argStr))
+		}
+
+		key := strings.TrimSpace(parts[0])
+		value := strings.TrimSpace(parts[1])
+		switch key {
+		case "max-length":
+			var err error
+			r.receiverNameMaxLength, err = strconv.Atoi(value)
+			if err != nil {
+				panic(fmt.Sprintf("Invalid value %s for the configuration key max-length, expected integer value: %v", value, err))
+			}
+		default:
+			panic(fmt.Sprintf("Unknown configuration key %s for %s rule.", key, r.Name()))
+		}
+	}
+}
```

Comment:
That's a function wide `if` statement

I would suggest this instead

```suggestion
func (r *ReceiverNamingRule) configure(arguments lint.Arguments) {
	r.Lock()
	defer r.Unlock()
	if r.receiverNameMaxLength > 0 {
		return
	}

	if len(arguments) < 1 {
		r.receiverNameMaxLength = defaultReceiverNameMaxLength
		return
	}
	arg := arguments[0]
	argStr, ok := arg.(string)
	if !ok {
		panic(fmt.Sprintf("Invalid argument for %s rule. Expecting a string, got %T", r.Name(), arg))
	}

	parts := strings.Split(argStr, "=")
	if len(parts) != 2 {
		panic(fmt.Sprintf("Invalid argument for %s rule. Expecting a string of the form 'key=value', got %s", r.Name(), argStr))
	}

	key := strings.TrimSpace(parts[0])
	value := strings.TrimSpace(parts[1])
	switch key {
	case "max-length":
		var err error
		r.receiverNameMaxLength, err = strconv.Atoi(value)
		if err != nil {
			panic(fmt.Sprintf("Invalid value %s for the configuration key max-length, expected integer value: %v", value, err))
		}
	default:
		panic(fmt.Sprintf("Unknown configuration key %s for %s rule.", key, r.Name()))
	}
}
```

Also I fixed the typo `an string` twice